bin/serverless

```diff
@@ -26,6 +28,16 @@ const invocationId = uuid.v4();
 
 // boot up error reporting via sentry before anything
 (() => initializeErrorReporter(invocationId).then(() => {
+  const requiredVersion = packageJson.engines.node;
+  const nodeVersion = process.version;
+  if (!semver.satisfies(nodeVersion, requiredVersion)) {
+    const msg = [
+      `Serverless Framework requires at least Node.js version ${requiredVersion}. `,
+      `You use version ${nodeVersion}.`,
+    ].join('');
+    throw new Error(msg);
```

Comment:
@dschep I just went ahead and implemented such a check. However I feel like this would be certainly a breaknig change based on the conversation about npms way to treat the `engines` attribute in `package.json`.

IMHO we should remove this for now. This way old Node.js version might still "just work".

We can remove it in a subsequent commit so that we at least have the code in the history. What do you think?

Reply:
should I update #6074 with an env var or something to disable SFE support since it requires node >= 6? Now that i think of it, we probably should have that option anyway, right?

Reply:
Maybe leave it and log the message instead of throwing it as an error?

Reply:
I am afraid that once older version is dropped from CI, newer-JS feature such as destructuring and async/await will be easily used.
Then serverless no longer runnable in formerly-supported Node.js versions, which is breaking change from user perspective.
So logging warning mean little, I think.

From contributor/user point of view, I think this deprecating older Node.js  is good timing for major version up, so that many rejected-in-v1 features can be introduced to increase DX.
v1 is already coming up to 3 years old.

Reply:
good point, @exoego. If old versions are no longer tested, they'll be broken soon enough.

Reply:
Just updated the code so that it's not throwing an error but logging the message.

@dschep could you explain what problem this PR introduces with the SFE needs for #6074? Should we downgrade to Node 6 here?